ios/browser/api/password/brave_password_api.mm

```diff
@@ -0,0 +1,472 @@
+#include "brave/ios/browser/api/password/password_store_listener_ios.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+namespace brave {
+namespace ios {
+password_manager::PasswordForm::Scheme PasswordFormSchemeForPasswordFormDigest(
+    PasswordFormScheme scheme) {
+  switch (scheme) {
+    case PasswordFormSchemeTypeHtml:
+      return password_manager::PasswordForm::Scheme::kHtml;
+    case PasswordFormSchemeTypeBasic:
+      return password_manager::PasswordForm::Scheme::kBasic;
+    case PasswordFormSchemeTypeDigest:
+      return password_manager::PasswordForm::Scheme::kDigest;
+    case PasswordFormSchemeTypeOther:
+      return password_manager::PasswordForm::Scheme::kOther;
+    case PasswordFormSchemeUsernameOnly:
+      return password_manager::PasswordForm::Scheme::kUsernameOnly;
+    default:
+      return password_manager::PasswordForm::Scheme::kHtml;
+  }
+}
+
+PasswordFormScheme PasswordFormSchemeFromPasswordManagerScheme(
+    password_manager::PasswordForm::Scheme scheme) {
+  switch (scheme) {
+    case password_manager::PasswordForm::Scheme::kHtml:
+      return PasswordFormSchemeTypeHtml;
+    case password_manager::PasswordForm::Scheme::kBasic:
+      return PasswordFormSchemeTypeBasic;
+    case password_manager::PasswordForm::Scheme::kDigest:
+      return PasswordFormSchemeTypeDigest;
+    case password_manager::PasswordForm::Scheme::kOther:
+      return PasswordFormSchemeTypeOther;
+    case password_manager::PasswordForm::Scheme::kUsernameOnly:
+      return PasswordFormSchemeUsernameOnly;
+    default:
+      return PasswordFormSchemeTypeHtml;
+  }
+}
+}  // namespace ios
+}  // namespace brave
+
+#pragma mark - IOSPasswordForm
+
+@interface IOSPasswordForm () {
+  GURL gurl_;
+  std::string signon_realm_;
+  base::Time date_created_;
+  base::Time date_last_used_;
+  base::Time date_password_changed_;
+  std::u16string username_element_;
+  std::u16string username_value_;
+  std::u16string password_element_;
+  std::u16string password_value_;
+  password_manager::PasswordForm::Scheme password_form_scheme_;
+}
+@end
+
+@implementation IOSPasswordForm
+
+- (instancetype)initWithURL:(NSURL*)url
+                signOnRealm:(NSString*)signOnRealm
+                dateCreated:(NSDate*)dateCreated
+               dateLastUsed:(NSDate*)dateLastUsed
+        datePasswordChanged:(NSDate*)datePasswordChanged
+            usernameElement:(NSString*)usernameElement
+              usernameValue:(NSString*)usernameValue
+            passwordElement:(NSString*)passwordElement
+              passwordValue:(NSString*)passwordValue
+            isBlockedByUser:(bool)isBlockedByUser
+                     scheme:(PasswordFormScheme)scheme {
+  if ((self = [super init])) {
+    [self setUrl:url];
+
+    [self setSignOnRealm:signOnRealm];
+
+    if (dateCreated) {
+      [self setDateCreated:dateCreated];
+    }
```

Comment:
Any reason to check nil and call setter methods like this instead of just doing `self.dateCreated = dateCreated`? (same for all others)

Reply:
Null check are removed from initializer and placed in setter for nullable fields.
https://github.com/brave/brave-core/pull/12360/commits/f1ed2d6425c64f0f9a7d7075c7af10d5d38b04c5